c2corg_ui/static/js/imageuploader.js

```diff
@@ -278,31 +286,22 @@ app.ImageUploaderController.prototype.save = function() {
   var defer = this.q_.defer();
 
   this.api_.createImages(this.files, this.documentService.document)
-  .then(function() {
+  .then(function(data) {
     var meta;
     var id;
+    var images = data['config']['data']['images'];
 
     $('.img-container').each(function(i, image) {
-      meta = this.files[i]['metadata'];
+      meta = images[i]['metadata'];
       id = 'image-' + (+new Date());
-      this.files[i]['document_id'] = id;
+      images[i]['document_id'] = id;
```

Comment:
well, in this case it has nothing to do with the 'real' document_id, it's a randomly generated on the fly just to render things correctly ui-side.